selfdrive/ui/ui.cc

```diff
@@ -58,6 +58,25 @@ void ui_init(UIState *s) {
   s->vipc_client = s->vipc_client_rear;
 }
 
+static int get_path_length_idx(const cereal::ModelDataV2::XYZTData::Reader &line, const float path_height) {
+  const auto line_x = line.getX();
+  int max_idx = 0;
+  for (int i = 0; i < TRAJECTORY_SIZE && line_x[i] < path_height; ++i) {
+    max_idx = i;
+  }
+  return max_idx;
+}
+
+static void update_lead(UIState *s, const cereal::RadarState::Reader &radar_state,
+                        const cereal::ModelDataV2::XYZTData::Reader &line, int idx) {
+  auto &lead_data = s->scene.lead_data[idx];
+  lead_data = (idx == 0) ? radar_state.getLeadOne() : radar_state.getLeadTwo();
+  if (lead_data.getStatus()) {
+    const int path_idx = get_path_length_idx(line, lead_data.getDRel());
+    car_space_to_full_frame(s, lead_data.getDRel(), lead_data.getYRel(), line.getZ()[path_idx], &s->scene.lead_vertices[idx]);
```

Comment:
```suggestion
    car_space_to_full_frame(s, lead_data.getDRel(), lead_data.getYRel(), -line.getZ()[path_idx], &s->scene.lead_vertices[idx]);
```

Compare to lane lines:
`    v += car_space_to_full_frame(s, line_x[i], -line_y[i] - y_off, -line_z[i] + z_off, v, margin);`

I would expect y to be also wrong, but looks like radard uses a different convention than the model.